src/azure-cli/azure/cli/command_modules/rdbms/validators.py

```diff
@@ -749,6 +749,18 @@ def validate_and_format_restore_point_in_time(restore_time):
                               "Please use ISO format e.g., 2021-10-22T00:08:23+00:00.")
 
 
+def is_citus_cluster(cmd, resource_group_name, server_name):
+    server_operations_client = cf_postgres_flexible_servers(cmd.cli_ctx, '_')
+    server = server_operations_client.get(resource_group_name, server_name)
+
+    return server.cluster and server.cluster.cluster_size > 0
+
+
+def validate_citus_cluster(cmd, resource_group_name, server_name):
```

Comment:
You can add the validator to param definition instead of calling it in each custom function. For example https://github.com/Azure/azure-cli/blob/14084adeae203a3f3342cab81cd3c75ec16e2ac4/src/azure-cli/azure/cli/command_modules/rdbms/_params.py#L469-L473
validator definition: https://github.com/Azure/azure-cli/blob/14084adeae203a3f3342cab81cd3c75ec16e2ac4/src/azure-cli/azure/cli/command_modules/rdbms/validators.py#L783-L788

Reply:
@evelyn-ys This can be useful for some arguments maybe within the create or update command, however we need entire commands blocked from use.

Example: a cluster should not have any access to manage replicas. Which is why the custom commands dealing with replica all have the validator.